pkg/prometheus/promcfg.go

```diff
@@ -626,9 +626,24 @@ func (cg *ConfigGenerator) Generate(
 		cfg = append(cfg, cg.generateRemoteReadConfig(p, store))
 	}
 
+	cfg = cg.appendTSDBConfig(cfg, p.Spec.TSDB)
+
 	return yaml.Marshal(cfg)
 }
 
+func (cg *ConfigGenerator) appendTSDBConfig(cfg yaml.MapSlice, tsdb v1.TSDBSpec) yaml.MapSlice {
+	if tsdb.OutOfOrderTimeWindow == "" {
+		return cfg
+	}
+
+	return cg.WithMinimumVersion("2.39.0").AppendMapItem(cfg, "tsdb", yaml.MapSlice{
+		{
```

Comment:
Should we log warning to user if lower prometheus version is used and `OutOfOrderTimeWindow` specified?

Reply:
it's already taken care of by `cg.WithMinimumVersion("2.39.0")` :)

Reply:
aah thanks :)